### PR TITLE
doc fix for config-changed

### PR DESCRIPTION
**Pull request by PietroPasotti**
Config-changed docfix

## Documentation changes

Already updated https://discourse.charmhub.io/t/config-changed-event/6465

## Changelog
- Fixed config-changed docstring.

### CODE REVIEW

File: ops/charm.py [hunk 1]
```diff
@@ -148,18 +148,20 @@ class RemoveEvent(HookEvent):
...
-    - after a :class:`relation is created <RelationCreatedEvent>`.
-    - after a :class:`leader is elected <LeaderElectedEvent>`.
-    - after changing charm configuration using the GUI or command line
-      interface
-    - when the charm :class:`starts <StartEvent>`.
-    - when a new unit :class:`joins a relation <RelationJoinedEvent>`.
-    - when there is a :class:`change to an existing relation <RelationChangedEvent>`.
+    """Event triggered when a configuration change occurs.
+
+    This event can fire in several situations.
```
Comment (rwcarlsen) [inline ops/charm.py:153]:
```suggestion
    This event can fire in several situations:
```

**Review by pengale — APPROVED** [review]

File: ops/charm.py [hunk 2]
```diff
@@ -148,18 +148,20 @@ class RemoveEvent(HookEvent):
...
+    - Right between a :class:`install <InstallEvent>` and a
+      :class:`starts <StartEvent>` during the startup sequence
+      (when you first deploy a unit). This event notifies the charm of
+      its initial configuration.
+    - When the cloud admin reconfigures the charm via the juju CLI, i.e.
+      `juju config my-charm foo=bar`. This event notifies the charm of
+      its new configuration. (The event itself, however, is now aware of *what*
+      specifically has changed in the config).
+    - When networking changes (if the machine reboots and comes up
+      with a different IP).
```
[the remark this answers is not in the document]
Comment (sed-i) [reply ops/charm.py:164]: config-changed fires also on upgrade and on churn (`kubectl delete`).

[the remark this answers is not in the document]
Comment (sed-i) [reply ops/charm.py:164]: Agreed, but would need to clarify what "startup" really means. Personally, when I think of "startup", I think of all the `install` cascade.

Consider the following:
- a unit is rescheduled on a different cluster (StoredState scratched)
- this particular "charm" (as in code snapshot) never saw/acted on an `install`, but only config-changed, upgrade-charm and pebble-ready.

Does the above still qualify as "startup"? In some sense yes, but it's tricky.

Would it be obvious from these docs that config-changed is guaranteed to fires on refresh? Probably not.

**Review by rwcarlsen — APPROVED** [review]

File: ops/charm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply ops/charm.py]: I gave a second shot at making it clear that config-changed is not specifically bound to the startup sequence. Does this help?

**Review by jnsgruk — APPROVED** [review]

File: ops/charm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply ops/charm.py]: @jnsgruk seems to think so, good enough for me :)

**Pull request merged**